### PR TITLE
feat(schemas) Make Columns and ColumnnType generic and parametric with regards to the modifier types.

### PR DESCRIPTION
**Pull request by fpacifici**
Followup to https://github.com/getsentry/snuba/pull/1439.

This makes ColumnType, Column and ColumnSet generic so we can impose that schemas and migrations will use two separate sets of modifiers.

### CODE REVIEW

File: snuba/migrations/parse_schema.py [hunk 1]
```diff
@@ -59,20 +59,21 @@
 )
 
 
-def merge_modifiers(col_type: ColumnType, modifiers: MigrationModifiers) -> ColumnType:
+def merge_modifiers(
+    col_type: ColumnType[MigrationModifiers], modifiers: MigrationModifiers
+) -> ColumnType[MigrationModifiers]:
     # TODO: Remove these assertions when ColumnType will be generic
```
Comment (evanh) [inline snuba/migrations/parse_schema.py:65]: This comment seems like it can go.

**Review by evanh — APPROVED** [review]

File: snuba/clickhouse/columns.py [hunk 2]
```diff
@@ -429,23 +439,23 @@ def _repr_content(self) -> str:
...
+    def set_modifiers(self, modifiers: Optional[TModifiers]) -> Enum[TModifiers]:
         return Enum(values=self.values, modifiers=modifiers)
 
-    def get_raw(self) -> Enum:
+    def get_raw(self) -> Enum[TModifiers]:
         return Enum(self.values)
 
 
-class ColumnSet:
+class ColumnSet(Generic[TModifiers]):
```
Comment (lynnagara) [inline snuba/clickhouse/columns.py:458]: This class doesn't need to be generic since it's only used in schemas and not migrations

Comment (fpacifici) [reply snuba/clickhouse/columns.py:458]: you are right, I did not notice that. Let me remove this.

Comment (lynnagara) [reply snuba/clickhouse/columns.py:458]: thanks

**Pull request merged**